src/python/pants/backend/helm/dependency_inference/chart.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import logging
+
+from pants.backend.helm.resolve import artifacts
+from pants.backend.helm.resolve.artifacts import ThirdPartyArtifactMapping
+from pants.backend.helm.subsystems import helm
+from pants.backend.helm.subsystems.helm import HelmSubsystem
+from pants.backend.helm.target_types import AllHelmChartTargets, HelmChartMetaSourceField
+from pants.backend.helm.util_rules.chart import HelmChartMetadata
+from pants.engine.addresses import Address
+from pants.engine.internals.selectors import Get
+from pants.engine.rules import collect_rules, rule
+from pants.engine.target import InferDependenciesRequest, InferredDependencies
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+from pants.util.ordered_set import OrderedSet
+from pants.util.strutil import pluralize
+
+logger = logging.getLogger(__name__)
+
+
+class InferHelmChartDependenciesRequest(InferDependenciesRequest):
+    infer_from = HelmChartMetaSourceField
+
+
+@rule(desc="Inferring Helm chart dependencies", level=LogLevel.DEBUG)
+async def infer_chart_dependencies_via_metadata(
+    request: InferHelmChartDependenciesRequest,
+    all_chart_tgts: AllHelmChartTargets,
+    third_party_mapping: ThirdPartyArtifactMapping,
+    subsystem: HelmSubsystem,
+) -> InferredDependencies:
+    # Build a mapping between the available Helm chart targets and their names
+    first_party_chart_mapping: dict[str, Address] = {}
+    for tgt in all_chart_tgts:
+        first_party_chart_mapping[tgt.address.target_name] = tgt.address
```

Comment:
This logic only relies on `AllHelmChartTargets` but will be performed for each inference request. Maybe move to a separate rule that generates a "FirstPartyChartMapping" type? Similar to what has already been done in this PR for `ThirdPartyArtifactMapping`.